src/Database/Database.php

```diff
@@ -382,6 +383,20 @@ public function createAttribute(string $collection, string $id, string $type, in
     {
         $collection = $this->getCollection($collection);
 
+        // attribute IDs are case insensitive
+        $attributes = $collection->getAttribute('attributes', []); /** @var Document[] $attributes */
+        \array_walk($attributes, function ($attribute) use ($id) {
+            if ($attribute instanceof Document) {
+                if (\strtolower($attribute->getId()) === \strtolower($id)) {
+                    throw new DuplicateException('Attribute already exists', 400);
+                }
+            } else {
+                if (\strtolower($attribute['$id']) === \strtolower($id)) {
+                    throw new DuplicateException('Attribute already exists', 400);
+                }
+            }
+        });
```

Comment:
```suggestion
        foreach($attributes as $attribute) {
                if (\strtolower($attribute->getId()) === \strtolower($id)) {
                    throw new DuplicateException('Attribute already exists');
                }
            }
        }
```

1. I think the behavior should be consistent now with this fix: https://github.com/utopia-php/database/pull/53/files and we can drop the instance check - please confirm.
2. Using HTTP status codes for errors in this context is useless.
3. Foreach syntax is easier to read, and has better performance: http://www.ktorides.com/2015/02/php-array_walk-vs-foreach/